src/healthRecord/healthRecord.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@nestjs/common';
+import { HealthColumn } from 'src/entities/healthColumn.entity';
+import { HealthData } from 'src/entities/healthData.entity';
+import { HealthRecord } from 'src/entities/healthRecord.entity';
+import { getManager } from 'typeorm';
+import { HealthDataDto, HealthRecordDto } from './dto/healthRecord.dto';
+
+@Injectable()
+export class HealthRecordService {
+  constructor() {}
+
+  async getHealthData(uid: number, healthRecordName: string): Promise<HealthRecordDto> {
+    const healthDataQuery = getManager()
+      .createQueryBuilder()
+      .select('hr.hrName', 'hrName')
+      .addSelect('hc.columnName', 'columnName')
+      .addSelect('hc.unit', 'unit')
+      .addSelect('hd.value', 'value')
+      .addSelect("DATE_FORMAT(hd.timestamp, '%Y-%m-%d %H:00:00')", 'timestamp')
+      .from(HealthRecord, 'hr')
+      .innerJoin(HealthColumn, 'hc', 'hr.uid = hc.uid AND hr.hrName = hc.hrName')
+      .innerJoin(HealthData, 'hd', 'hc.columnName = hd.ColumnName AND hc.hrName = hd.hrName AND hc.uid = hd.uid')
+      .where('hr.uid = :uid', { uid: uid })
+      .andWhere('hr.hrName = :hrName', { hrName: healthRecordName })
+      .orderBy("DATE_FORMAT(hd.timestamp, '%Y-%m-%d %H:00:00')");
+
+    const healthDataRaw = await healthDataQuery.getRawMany();
+    const distinctValueByColumns = this.extractDistinctValueByColumns(healthDataRaw, ['columnName', 'unit']);
+
+    const columnNames = [];
+    const units = [];
+    distinctValueByColumns.map((d) => {
+      columnNames.push(d.columnName);
+      units.push(d.unit);
+    });
+
+    const healthData = this.healthDataFormatter(healthDataRaw, columnNames);
+    const result = {
+      tableName: healthRecordName,
+      columnNames: columnNames,
+      units: units,
+      data: healthData,
+    };
+
+    return result;
+  }
+
+  private healthDataFormatter(healthDataRaw, columnNames): HealthDataDto[] {
+    const columnNumbers = columnNames.length;
+    const healthData = [];
+    healthDataRaw.map((h) => {
```

Comment:
Consider using `.forEach` or a normal for loop instead of `.map` since it returns a new array which you don't seem to need.